src/bundle.ts

```diff
@@ -14,8 +14,24 @@ const _bundle = (document: unknown, path: string, cur?: unknown) => {
     if (property === '$ref' && typeof propertyValue === 'string' && isLocalRef(propertyValue)) {
       const _path = pointerToPath(propertyValue);
       const bundled$Ref = get(document, _path);
-      const exists = !!get(objectToBundle, _path);
+      const exists = has(objectToBundle, _path);
       if (bundled$Ref && !exists) {
+        const pathProcessed = [];
+
+        // make sure arrays and object decisions are preserved when copying over the portion of the tree
+        for (const key of _path) {
+          pathProcessed.push(key);
```

Comment:
Shouldn't this happen after? Since the name says `processed` it should be added after the code has been executed?

Reply:
I can re-name the property, would that help? It happens first, because we use it to look up the value at the path right after.

Reply:
Nah don't bother, it's fine.